code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -54,13 +54,20 @@
 
 /obj/item/reagent_containers/glass/throw_impact(atom/hit_atom, var/speed)
 	. = ..()
-	if(speed > fragile && !no_shatter)
+	if(no_shatter)
+		return
+	if(speed > fragile)
 		shatter()
+		return
+	if  (speed <= fragile)
+		if (prob(50))
+			shatter()
```

Comment:
Seems a bit extreme to break with a 50/50 chance no matter the speed. Why not make this a function based on how fragile the item is?

Reply:
The % chance is now 20x the difference between hardness and throw speed.

Reply:
Making a function based on the fragility was extremely awkward and difficult because one was basically a negative value (more fragile means weaker) and the other a positive value (more throw force means more power) so I rewrote the entire variable to be hardness instead of fragile, and now objects only break absolutely if you throw hard enough to bypass their hardness, otherwise they have a change to break based on how hard you throw them.